Neos.Neos/Classes/Fusion/Helper/DimensionHelper.php

```diff
@@ -0,0 +1,202 @@
+     *
+     * @param Node $node
+     * @param string|ContentDimensionId $dimensionName
+     * @return string|null
+     */
+    public function findDefaultValue(Node $node, string|ContentDimensionId $dimensionName): ?string
+    {
+        $contentDimensionId = is_string($dimensionName) ? new ContentDimensionId($dimensionName) : $dimensionName;
+        $contentRepository = $this->contentRepositoryRegistry->get($node->subgraphIdentity->contentRepositoryId);
+        $rootValues = $contentRepository
+            ->getContentDimensionSource()
+            ->getDimension($contentDimensionId)
+            ?->getRootValues() ?: [];
+
+        $dimensionValue = reset($rootValues);
+
+        return $dimensionValue ? $dimensionValue->value : null;
+    }
+
+    /**
+     * Find all content dimensions in content repository defined by `node`.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.findDimensions(node)
+     *
+     * @param Node $node
+     * @return array<string,ContentDimension>
+     */
+    public function findDimensions(Node $node): array
+    {
+        $contentRepository = $this->contentRepositoryRegistry->get($node->subgraphIdentity->contentRepositoryId);
+
+        return $contentRepository->getContentDimensionSource()->getContentDimensionsOrderedByPriority();
+    }
+
+    /**
+     * Find the variant of `node` in the specified dimension and value.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.findVariantInDimension(node, 'language', 'en_UK')
+     *
+     * @param Node $node
+     * @param string|ContentDimensionId $dimensionName
+     * @param string|ContentDimensionValue $dimensionValue
+     * @return Node|null
+     */
+    public function findVariantInDimension(Node $node, string|ContentDimensionId $dimensionName, string|ContentDimensionValue $dimensionValue): ?Node
+    {
+        $contentDimensionId = is_string($dimensionName) ? new ContentDimensionId($dimensionName) : $dimensionName;
+        $contentDimensionValue = is_string($dimensionValue) ? new ContentDimensionValue($dimensionValue) : $dimensionValue;
+        $contentRepository = $this->contentRepositoryRegistry->get($node->subgraphIdentity->contentRepositoryId);
+
+        return $contentRepository
+            ->getContentGraph()
+            ->getSubgraph($node->subgraphIdentity->contentStreamId, $node->subgraphIdentity->dimensionSpacePoint->vary($contentDimensionId, $contentDimensionValue->value), $node->subgraphIdentity->visibilityConstraints)
+            ->findNodeById($node->nodeAggregateId);
+    }
+
+    /**
+     * Find all variants of `node` across the specified dimension.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.findVariantsInDimension(node, 'language')
+     *
+     * @param Node $node
+     * @param string|ContentDimensionId $dimensionName
+     * @return Nodes
+     */
+    public function findVariantsInDimension(Node $node, string|ContentDimensionId $dimensionName): Nodes
```

Comment:
where do we need this for? seems quite complex :)

Reply:
Can also be very handy for creating dimension menus.

Reply:
=> weg

Reply:
ToDo: Remove from helper and solve the edge use case in `Neos.Seo`